### ISSUE TITLE
"Attempted to beginRule does not match outer scope"

### ISSUE THREAD

**Issue by jukzi** [eclipse-platform/eclipse.platform#578]
org.eclipse.core.tests.internal.builders.ParallelBuildChainTest.testDependentBuildConfigsSubset()

silently logs
```
!MESSAGE Attempted to beginRule: org.eclipse.core.tests.internal.builders.TimerBuilder$1@49cc8332, does not match outer scope rule: R/
!STACK 0
java.lang.IllegalArgumentException: Attempted to beginRule: org.eclipse.core.tests.internal.builders.TimerBuilder$1@49cc8332, does not match outer scope rule: R/
	at org.eclipse.core.runtime.Assert.isLegal(Assert.java:66)
	at org.eclipse.core.internal.jobs.ThreadJob.illegalPush(ThreadJob.java:138)
	at org.eclipse.core.internal.jobs.ThreadJob.push(ThreadJob.java:402)
	at org.eclipse.core.internal.jobs.ImplicitJobs.begin(ImplicitJobs.java:66)
	at org.eclipse.core.internal.jobs.JobManager.beginRule(JobManager.java:343)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:240)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:303)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:392)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:395)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:506)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:454)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:536)
	at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:207)
	at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:300)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)

```

that sounds wrong.

**Comment by mickaelistria** [comment]
Has anything changed in the rules implementation lately? The test are really strongly relying on rules to determine what can be run in parallel and what cannot, and AFAIK they've been stable enough for a long time.

**Comment by jukzi** [comment]
i am not aware of any recent changes. may be it silently crashed since forever?

**Comment by mickaelistria** [comment]
Definitely not forever, and I don't think there is a reason for it to crash silently. However, as those are about build scheduling, if there is another build now running in parallel, it can invalidate the test.

**Comment by HeikoKlare** [comment]
Implicitly fixed by #641.
